src/main/scala/beam/router/skim/core/AbstractSkimmer.scala

```diff
@@ -58,7 +58,7 @@ abstract class AbstractSkimmerReadOnly extends LazyLogging {
   def getCurrentSkimValue(key: AbstractSkimmerKey): Option[AbstractSkimmerInternal] =
     Option(currentSkimInternal.get(key))
   def aggregatedFromPastSkims: Map[AbstractSkimmerKey, AbstractSkimmerInternal] = aggregatedFromPastSkimsInternal
-  def pastSkims: Map[Int, collection.Map[AbstractSkimmerKey, AbstractSkimmerInternal]] = pastSkimsInternal.toMap
+  def pastSkims: collection.Map[Int, collection.Map[AbstractSkimmerKey, AbstractSkimmerInternal]] = pastSkimsInternal
```

Comment:
Same as above - we shouldn't pass the internal directly..

Reply:
I restored that toMap and minimized the number of calls to this method.